### ISSUE TITLE
5.0 RC doesn't handle docker-image correctly in tasks

### ISSUE THREAD

**Issue by ralekseenkov** [concourse/concourse#3295]
## Bug Report

We started playing with 5.0 RC and ran into an issue:

1) When deployed on Kubernetes any task with 'docker-image' resource type fails with:
    ```
    mount: can't find /sys/fs/cgroup in /proc/mounts
    resource script '/opt/resource/in [/tmp/build/get]' failed: exit status 1
    ```
2) When built from master and deployed locally via docker-compose up, the same task fails with:
    ```
    unsupported resource type
    ```

When you switch from 'docker-image'  to 'registry-image', both errors disappear. It was quite difficult to figure out the reason, especially given that we saw 'cgroup' error first.

## Steps to Reproduce

Pipeline attached. The task that fails is 'rename-chart'.

## Expected Results

Pipelines with 'docker-image' resource type in tasks should continue to work after upgrading to 5.0. , but a deprecation warning should be shown.

If compatibility with 'docker-image' is dropped in 5.0, then it should be very clear it's not going to work anymore.

## Actual Results

Pipelines stop working.

## Version Info

* Concourse version: 5.0 RC
* Deployment type (BOSH/Docker/binary): Local from code, and Kubernetes
* Did this used to work? Yes

**Comment by ralekseenkov** [comment]
[pipeline_with_embedded_tasks.txt](https://github.com/concourse/concourse/files/2862588/pipeline_with_embedded_tasks.txt)

**Comment by cirocosta** [comment]
Oh, that's true, nice catch!

Just reproduced here with this smaller pipeline running 5.0:

```yaml
resources:
- name: image
  type: docker-image
  source: {repository: busybox}

jobs:
- name: test
  plan: [{get: image}]
```

I'm wondering if this is specific to the mounts that k8s is performing by default ... Did this use to work with 4.2.2 using the Helm chart? I don't have access to a cluster with 4.2.2 right now.

Interesting that after you hijack to the container, you can see that `/proc/mounts` reports the presence of the cgroup mount over there:

```
cgroup /sys/fs/cgroup tmpfs rw,relatime,mode=755 0 0
```

although you can't find the subsystems like you can in a non-k8s deploy:

```
cgroup /sys/fs/cgroup tmpfs rw,relatime,mode=755 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,relatime,blkio 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,relatime,memory 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,relatime,freezer 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,relatime,hugetlb 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,relatime,pids 0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,relatime,rdma 0 0
```

🤔 

---

Marking it as `k8s` so we can prioritize the work on that soon!

Thx!

**Comment by marco-m** [comment]
As an additional data point, `docker-image` works correctly with 5 RC deployed directly on VM (no Docker).

**Comment by vito** [comment]
I think this is related:

https://github.com/concourse/docker-image-resource/commit/fd492346246e9e9f61040bf31c740e80fccc9365

We haven't shipped a version of the Docker Image resource with this fix, though.

**Comment by cirocosta** [comment]
Oh, interesting, I just bumped the version of `hush-house` (k8s based using the latest RC image - which contains the newly released `docker-image-resource` and it worked 💯 

https://hush-house.concourse-ci.org/teams/main/pipelines/test/jobs/test/builds/6#L5c66178d:9:12

Could you give it a try @ralekseenkov ?

Thx!

**Comment by vito** [comment]
Yep - forgot to check back, I shipped Docker Image resource v1.2.0 which is now in 5.0-rc.79.

**Comment by vito** [comment]
This should be fixed in the latest RC - closing!
